operator/aws.go

```diff
@@ -249,8 +252,22 @@ func (o *AWSOperator) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Res
 		return ctrl.Result{}, o.removeFromVault(req.Namespace, req.Name)
 	}
 
+	// check if default-sts-ttl is set if not use config default
+	defaultTTL := o.DefaultTTL
+	if v, ok := serviceAccount.Annotations[defaultStsTTLAnnotation]; ok {
+		defaultTTL, err = time.ParseDuration(v)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error parsing default_sts_ttl %w", err)
+		}
+
+		// check new default ttl value is higher then min ttl set in config
+		if defaultTTL < o.MinTTL {
+			return ctrl.Result{}, fmt.Errorf("minimum default-sts-ttl value allowed is %s, its set to %s", o.MinTTL, defaultTTL)
```

Comment:
I think in this case we could log a warning message, but still set the minimum allowed ttl, instead of returning an error?

Reply:
No, I think it should error rather that "silently" doing something else.